### PR TITLE
fix: plumb ctx into asset-groups db methods

### PR DESCRIPTION
**Pull request by mistahj67**
## Description

Plumb context into gorm asset-group db methods
https://gorm.io/docs/context.html#Context-Timeout

I also noticed there was a number of unused db methods related to the asset-groups so I removed those to keep things nice and clean. (I didn't check the rest of the methods, but I do think it would be beneficial to give them all a quick audit)

## Motivation and Context

Currently as it stands any endpoints that rely on purely gorm do not respect `request.Context` and as such timeouts are not respected either. While we want to eventually get to the removal of the gorm dependency altogether, that is a long way and a lot of effort away and this I think this approach would give us that much room to maneuver while we get there.

## How Has This Been Tested?

Using a rest client and the prefer header
Add a sleep inside of the handler
Look for the `500 request timed out` after exceeding the timeout. (Note: the request will take as long as the sleep is set, it's not tied to the prefer header)

## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [x] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: cmd/api/src/go.mod [hunk 1]
```diff
@@ -47,6 +47,7 @@ require (
 	github.com/unrolled/secure v1.13.0
 	github.com/zenazn/goji v1.0.1
 	go.uber.org/mock v0.2.0
+	go.uber.org/zap v1.13.0
```
Comment (juggernot325) [inline cmd/api/src/go.mod:50]: It's curious how this new dependency got added. It doesn't look like it's used in any of the files in this PR.

Comment (zinic) [reply cmd/api/src/go.mod:50]: Let's `zap` this dependency if we aren't using it. Pun intended. Suffer.

**Review by zinic — CHANGES_REQUESTED** [review]
Requesting dependency clean up, otherwise changeset looks good.

File: cmd/api/src/go.mod (hunk 1 above)
Comment (mistahj67) [reply cmd/api/src/go.mod]: image

**Pull request merged**